### PR TITLE
Updated readme and action file

### PR DESCRIPTION
**Pull request by sarvsav**
This pull request updates the action file with proper keywords and documentation is updated
based on the practices followed by `google-github-actions` organization,

- [x] Tested locally

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,39 +6,54 @@
...
+      ## checkout the code from the repository
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Hello world action step
-        uses: go-feature-flag/gofeatureflag-lint-action@__hash_value__
-        id: hello          
+      
+      ## run the gofeatureflag-lint action
+      - name: Lint the config file
+        uses: go-feature-flag/gofeatureflag-lint-action@ea0524d562003105471e2ccdf3485e98db23fa48
```
Comment (thomaspoignant) [inline README.md]: I will suggest in the doc to add `v1` to prepare for the release when the tag will be ready.

```suggestion
        uses: go-feature-flag/gofeatureflag-lint-action@v1
```

File: README.md [hunk 2]
```diff
@@ -6,39 +6,54 @@
...
+
+        ## set the input parameters for the action
         with:
-          config-root: ${GITHUB_WORKSPACE}
+          flagFilePath: ${{ github.workspace }}/config/flag-config.yaml
+          flagFiletype: 'yaml'
```
Comment (thomaspoignant) [inline README.md]: All examples of Github Action I see around are using [kebab case](https://www.theserverside.com/definition/Kebab-case) for parameters.

I will change this into something like this:
```suggestion
          flag-file: ${{ github.workspace }}/config/flag-config.yaml
          format: yaml
```

File: README.md [hunk 3]
```diff
@@ -6,39 +6,54 @@
...
 ```
 
+## Inputs ⚙️
+
+| **Name** | **Required** | **Defaults**  | **Description**  |
+|:-----:|:-----:|:-----:|:-----:|
+|  flagFilePath     | false      | `config/flag-config.yaml`  | Absolute or relative path of the file with name  |
+|   flagFiletype    | false      | `yaml`  | config file of type `yaml`, `toml`, or `json`  |
```
Comment (thomaspoignant) [inline README.md]:
```suggestion
|  `flag-file`     | false      | `config/flag-config.yaml`  | Absolute or relative path of the file with name  |
|   `format`    | false      | `yaml`  | config file of type `yaml`, `toml`, or `json`  |
```

File: README.md [hunk 4]
```diff
@@ -6,39 +6,54 @@
...
+
+## Outputs ✉️
+
+| **Name** | **Description**  |
+|:-----:|:-----:|
+|  lintMessage     | `empty` if no error is found, else the value will be the error message  |
```
Comment (thomaspoignant) [inline README.md]:
```suggestion
|  lint-message     | `empty` if no error is found, else the value will be the error message  |
```

File: README.md [hunk 5]
```diff
@@ -6,39 +6,54 @@
...
+
+## Share the Love ❤️
+
+Like this project? Please give it a ★ on our GitHub!
```
Comment (thomaspoignant) [inline README.md:55]: Love this ❤️

File: action.yml [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
 
 inputs: # inputs of the action used as `args` in `lint`
-  configFilePath:  # path to the config file
-    description: 'Root directory of the config file'
-    required: true # make the input required
-    default: './config'
-  configFiletype: # filetype of the config file can be yaml, toml, or json
-    description: 'Filetype of the config file'
-    required: true # make the input required
+  flagFilePath:  # path to the flag file
```
Comment (thomaspoignant) [inline action.yml]:
```suggestion
  flag-file:  # path to the flag file
```

File: action.yml [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
+    description: 'Absolute or relative path to the flag file'
+    required: false
+    default: './config/flags.yaml' # default path is './config/flags.yaml
+  flagFiletype: # filetype of the flag file can be yaml, toml, or json
```
Comment (thomaspoignant) [inline action.yml]:
```suggestion
  format: # format of the flag file can be yaml, toml, or json
```

File: action.yml [hunk 8]
```diff
@@ -26,5 +27,9 @@ runs:
     
     ## run the linter script and capture the output
     - id: lint
-      run: run-lint.sh ${{ inputs.configFilePath }} ${{ inputs.configFiletype }}
+      run: run-lint.sh ${{ inputs.flagFilePath }} ${{ inputs.flagFiletype }}
```
Comment (thomaspoignant) [inline action.yml]:
```suggestion
      run: run-lint.sh ${{ inputs.flag-file }} ${{ inputs.format }}
```

File: action.yml [hunk 9]
```diff
@@ -1,14 +1,15 @@
...
+  flagFilePath:  # path to the flag file
+    description: 'Absolute or relative path to the flag file'
+    required: false
+    default: './config/flags.yaml' # default path is './config/flags.yaml
+  flagFiletype: # filetype of the flag file can be yaml, toml, or json
+    description: 'Filetype of the flag file'
+    required: false 
     default: 'yaml' # default file format is yaml
 
 outputs:
```
Comment (thomaspoignant) [inline action.yml:15]: we should change `lintMessage` to be consistent with the rest of kebab case.

**Review by thomaspoignant — CHANGES_REQUESTED** [review]
Some remarks about the naming of the params.
The rest looks fine to me.

File: README.md (hunk 1 above)
Comment (sarvsav) [reply README.md]: Yes, I agree and have updated the example.

File: README.md (hunk 2 above)
Comment (sarvsav) [reply README.md]: I was using the kebab-case style, then later I tried to follow more Go style. I will update all the occurrence to kebab-case.

File: README.md (hunk 3 above)
Comment (sarvsav) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (sarvsav) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (sarvsav) [reply README.md:55]: Thanks 🙏

File: action.yml (hunk 6 above)
Comment (sarvsav) [reply action.yml]: Done

File: action.yml (hunk 7 above)
Comment (sarvsav) [reply action.yml]: Done

File: action.yml (hunk 8 above)
Comment (sarvsav) [reply action.yml]: Done

File: README.md [hunk 10]
```diff
@@ -3,42 +3,57 @@
 <p align="center">
     <img width="250" height="238" alt="Go Feature Flag" src="https://github.com/thomaspoignant/go-feature-flag/raw/main/logo.png" />
     <h2 align="center">GoFeatureFlagLint</h2>
-    <p align="center">Lint your feature flag config files.</p>
+    <p align="center">Lint your feature flag config files</p>
 </p>
 
-This action will perform the linting of the gofeatureflag config file.
+The `go-feature-flag/gofeatureflag-lint-action` Github action performs the
+linting of the [gofeatureflag](https://gofeatureflag.org/) flag config file. It
```
Comment (thomaspoignant) [inline README.md]:
```suggestion
linting of the [GO Feature Flag](https://gofeatureflag.org/) flag config file. It
```

**Review by thomaspoignant — DISMISSED** [review]

**Review by thomaspoignant — APPROVED** [review]

**Pull request merged**